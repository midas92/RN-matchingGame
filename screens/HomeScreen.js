import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import FlippableCard from '../components/FlippableCard';

const HomeScreen = ({ 
  navigation 
}) => {
  return (
    <View style={{flex: 1}}>
      <View style={styles.textContainer}>
        <Text style={styles.title}>
          Welcome to Card Flipper!
        </Text>
      </View>
      <View style={styles.buttonContainer}>
        <FlippableCard/>
        <TouchableOpacity
            onPress={() => 
              navigation.navigate('SetupGameScreen')
            }
            style={styles.playButton}>
          <Text style={styles.playButtonText}>Play</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  textContainer: {
    flex: 2,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
  
  title: {
    fontSize: 30
  },
  
  buttonContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#fff',
  },

  playButton: {
    height: 50,
    width: 150,
    
    alignItems: 'center',
    justifyContent: 'center',
    
    shadowColor: 'black',
    shadowRadius: 2,
    shadowOpacity: 1,
    backgroundColor: '#34a8eb',
    borderRadius: 5,
  },
  playButtonText: {
    fontSize: 20
  },
});

export default HomeScreen;