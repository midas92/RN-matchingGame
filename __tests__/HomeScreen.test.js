import React from 'react'
import { render, fireEvent } from '@testing-library/react-native';
import HomeScreen from '../screens/HomeScreen';

const navigation = { navigate: jest.fn() };

describe('<HomeScreen/>', () => {
  it('should match snapshot', () => {
    const result = render(<HomeScreen navigation={navigation}/>).toJSON();
    expect(result).toMatchSnapshot();
  });

  it('should render the HomeScreen component', () => {
    const result = render(<HomeScreen navigation={navigation}/>);
    expect(result).toBeTruthy();
  });

  it('should navigate to setup game screen on press of Play', () => {
    const result = render(<HomeScreen navigation={navigation}/>);
    const playButton = result.getByTestId('playButton');

    fireEvent(playButton, 'press');

    expect(navigation.navigate).toBeCalledWith('SetupGameScreen');
  });
  
  it('should navigate to high score screen on press of High Scores', () => {
    const result = render(<HomeScreen navigation={navigation}/>);
    const scoreScreenButton = result.getByTestId('scoresButton');

    fireEvent(scoreScreenButton, 'press');

    expect(navigation.navigate).toBeCalledWith('ScoreScreen');
  });
});
