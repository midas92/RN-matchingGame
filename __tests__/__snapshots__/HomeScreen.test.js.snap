// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen/> should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 30,
        }
      }
    >
      Welcome to Card Flipper!
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
    testID="test"
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#34a8eb",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "shadowColor": "black",
          "shadowOpacity": 1,
          "shadowRadius": 2,
          "width": 150,
        }
      }
      testID="playButton"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Play
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#34a8eb",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "shadowColor": "black",
          "shadowOpacity": 1,
          "shadowRadius": 2,
          "width": 150,
        }
      }
      testID="scoresButton"
    >
      <Text
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        High Scores
      </Text>
    </View>
  </View>
</View>
`;
